fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePlanner.java

```diff
@@ -51,45 +54,143 @@ public LucenePlanner(@Nonnull final RecordMetaData metaData, @Nonnull final Reco
         super(metaData, recordStoreState, indexTypes, timer);
     }
 
-    private ScanComparisons getScanForAndLucene(@Nonnull AndComponent filter, @Nullable FilterSatisfiedMask filterMask) {
+    private Comparisons.LuceneComparison getScanForFieldWithComparison(@Nullable String parentFieldName, @Nonnull FieldWithComparison filter,
+                                                               @Nullable FilterSatisfiedMask filterSatisfiedMask) {
+        String completeFieldName = filter.getFieldName();
+        Comparisons.Comparison comparison =  filter.getComparison();
+        if (parentFieldName != null) {
+            completeFieldName = parentFieldName + "." + completeFieldName;
+        }
+        String comparisonString;
+        Comparisons.Type type = filter.getComparison().getType();
+        switch (type) {
+            case EQUALS:
+            case TEXT_CONTAINS_ALL:
+            case TEXT_CONTAINS_PHRASE:
+                comparisonString = "%s:";
+                break;
+            case IS_NULL:
+                comparisonString = "NOT %s";
+                break;
+            case NOT_EQUALS:
+                comparisonString = "NOT %s:";
+                break;
+            case TEXT_CONTAINS_ANY:
+            case TEXT_CONTAINS_PREFIX:
+            default:
+                return null;
+        }
+        if (type != Comparisons.Type.IS_NULL) {
+            if (comparison.getComparand() instanceof String) {
+                comparisonString = comparisonString + "\"%s\"";
```

Comment:
I believe that there are a number of special characters that need escaping. I'm okay with deferring that to a follow-up.

Reply:
Well, yes we'll have to escape some of the Lucene query characters like + and ? and (. For now I think this is fine since the parser will probably disallow it in another location.